modules/imgproc/src/opencl/boxFilter3x3.cl

```diff
@@ -0,0 +1,127 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+__kernel void boxFilter3x3_8UC1_cols16_rows2(__global const uint* src, int src_step,
+                                             __global uint* dst, int dst_step, int dst_offset, int rows, int cols
```

Comment:
dst_offset is removed from kernel arg and is checked in the host code.

for src_step/dst_step, checks for (type == CV_8UC1) and (_src.cols() % 16 == 0) should tell that src_step % 4 == 0 implicitly, so in the update version, I added check (_src.step() == _dst.step()) in the host code to make sure dst_step is multiple of 4 as well. pls review updated patch.